pkg/types/var.go

```diff
@@ -135,6 +149,45 @@ func (vs *VarSet) Merge(v Var) error {
 	return nil
 }
 
+// AbsorbSet absorbs other vars with error on (name,value) collision.
+func (vs *VarSet) AbsorbSet(incoming VarSet) error {
+	for _, v := range incoming.set {
+		vs.Absorb(v)
+	}
+	return nil
+}
+
+// AbsorbSlice absorbs a Var slice with error on (name,value) collision.
+// Empty fields in incoming vars are defaulted.
+func (vs *VarSet) AbsorbSlice(incoming []Var) error {
+	for _, v := range incoming {
+		if err := vs.Absorb(v); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Absorb absorbs another Var with error on (name,value) collision.
+// Empty fields in incoming Var is defaulted.
+func (vs *VarSet) Absorb(v Var) error {
```

Comment:
Is this function going to replace `Merge`? Similarly, are other functions going to replace `MergeSlice` and `MergeSet`?

Reply:
@Liujingfang1 I think that long term we should get MergeSlice and MergeSet to be replaced by AborbSlice and AbsorbSet. MergeSlice just errors if something is trying to import a variable with the exact same name. AbsorbSlice errors if something is trying to import a variable with the same name but a different definition.

Reply:
Sounds good.